sale_order_invoicing_queued/models/queue_job.py

```diff
@@ -0,0 +1,17 @@
+# Copyright 2019 Tecnativa - Pedro M. Baeza
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+
+from odoo import models
+
+
+class QueueJob(models.Model):
+    _inherit = 'queue.job'
+
+    def cancel_sudo(self):
+        self.sudo().filtered(
```

Comment:
I think you have to use ACLs here that gives unlink access to `queue.job` records linked to a sale order if the user has read access to the sale order.

Otherwise, using sudo and unlink in a public method is a very dangerous operation. You can abuse the XMLRPC to do things you shouldn't be able to do.

Reply:
OK, good point, but no need to rework ACL. It's already good suited. Just removing the sudo.